packages/discovery-react-components/src/components/DiscoverySearch/types.ts

```diff
@@ -11,3 +11,18 @@ export type SearchParams = Omit<DiscoveryV2.QueryParams, 'projectId' | 'headers'
    */
   returnFields?: string[];
 };
+
+export interface DocumentProviderProps extends Pick<QueryResult, 'document_id' | 'metadata'> {
+  extracted_metadata?: JsonObject;
+}
+
+export interface DocumentProvider {
+  /**
+   * Return `true` if service can provide the given document
+   */
+  provides: (document: DocumentProviderProps) => boolean;
```

Comment:
I think we need to allow `async` way to provider functions. They may involve API calls.